src/SmoFacade/Database.cs

```diff
@@ -134,5 +117,31 @@ public void MultiUserMode()
       _database.DatabaseOptions.UserAccess = DatabaseUserAccess.Multiple;
       _database.Alter(TerminationClause.RollbackTransactionsImmediately);
     }
+
+    private void ThrowIfUnreadyToDrop()
+    {
+      if(Restoring)
+        return; // restoring state means we're good to drop
```

Comment:
If it's restoring, shouldn't we wait until it is finished?

Reply:
no - restoring state doesn't necessarily mean that it's currently writing a backup.  
Restoring state happens on secondary nodes when the DB is removed from the availability group. 

See:  https://docs.microsoft.com/en-us/sql/database-engine/availability-groups/windows/remove-a-primary-database-from-an-availability-group-sql-server?view=sql-server-ver15#FollowUp